HotelPriceScout/Pages/Index.razor

```diff
@@ -7,178 +7,217 @@
+                                    <td class="@(priceList.Last()==item ? "table-success rounded-bottom":"table-success")">@item.Price,-</td>
+                                </tr>
+                            }
+
+                            else
+                            {
+                                <tr>
+                                    <td class="col text-center"> @MarketPriceItem.Name<span class="oi oi-arrow-right text-center"></span></td>
+                                    <td class="table-info">@MarketPriceItem.Price,-</td>
+                                </tr>
+                            }
+                        }
+                    </table>
+                    }
             </div>
-        }
-    </div>
+            <div>Date: @Cal.DayClicked</div>
+        </div>}
+</div>
 
 @code {
+    int mPrice;
     public int AdultsPicked = 1;
     public class numberOfAdults
     {
         public int Adults { get; set; }
         public string Text { get; set; }
     }
+
     public List<numberOfAdults> Adults = new List<numberOfAdults>()
-        {
+    {
             new numberOfAdults(){ Adults= 1, Text= "1 adult" },
             new numberOfAdults(){ Adults= 2, Text= "2 adults" },
             new numberOfAdults(){ Adults= 3, Text= "4 adults" },
-        };
+    };
 
     Dashboard Cal = new Dashboard();
     IEnumerable<string> listOfHotels = Enumerable.Empty<string>();
 
     List<Prices> priceList;
     Prices MarketPriceItem;
+    IEnumerable<MarketPriceModel> CalendarMarketPrices;
+
     private void GenerateThermometer()
     {
-        priceList =  PriceMeterGenerator.PriceListGenerator(); // This needs to get its data from the database, as a list of Price objects
-        MarketPriceItem =  (PriceMeterGenerator.MarketFinder(priceList));
+        priceList = PriceMeterGenerator.PriceListGenerator(); // This needs to get its data from the database, as a list of Price objects
+        MarketPriceItem = (PriceMeterGenerator.MarketFinder(priceList));
         priceList.Sort();
     }
 
+    public async void OnClickUpdate()
+    {
+        CalendarMarketPrices = await (Cal.DisplayComparedPrices(Cal.ToDay.ToString("yyyy-MM-dd"), Cal.MonthEnd.ToString("yyyy-MM-dd")));
+    }
+
     protected override async Task OnInitializedAsync()
     {
         GenerateThermometer();
-        listOfHotels = await _db.LoadStaticHotelResources("total");
+
         Cal.CreateMonth();
 
-       
+        CalendarMarketPrices = await (Cal.DisplayComparedPrices(Cal.TempDate.ToString("yyyy-MM-dd"), Cal.MonthEnd.ToString("yyyy-MM-dd")));
+    }
+
+    public int MarketPriceDaySetter(int k)
+    {
+        //The time is set to 23:59:59 to ensure that no matter the time of loading the data, the current day will be correct 
+        if (new DateTime(Cal.Year, Cal.Month, k, 23, 59, 59) >= Cal.ToDay)
+        {
+            IEnumerable<MarketPriceModel> SingelMarketPrice = CalendarMarketPrices
```

Comment:
Bare en lille kommentar til stavefejl "SingleMarketPrice" i stedet for "SingelMarketPrice" 😄